src/views/CashlinkReceive.vue

```diff
@@ -0,0 +1,690 @@
+            NetworkClient.Instance.on(NetworkClient.Events.CONSENSUS_ESTABLISHED, () => this.consensus = 'established');
+            NetworkClient.Instance.on(NetworkClient.Events.CONSENSUS_LOST, () => this.consensus = 'lost');
+            this.consensus = NetworkClient.Instance.consensusState;
+
+            this.cashlink.networkClient = NetworkClient.Instance;
+        }
+    }
+
+    private async claim() {
+        // Start loading screen
+        this.statusState = StatusScreen.State.LOADING;
+        this.statusTitle = 'Claiming Cashlink';
+        this.statusStatus = 'Connecting to Nimiq...';
+
+        this.isClaiming = true;
+
+        this.cashlink!.claim(this.activeAccount!.userFriendlyAddress);
+
+        // Set up transaction-relayed listener, so we know when the tx has been sent
+        try {
+            await new Promise((resolve, reject) => {
+                NetworkClient.Instance.on(NetworkClient.Events.TRANSACTION_RELAYED, (tx: DetailedPlainTransaction) => {
+                    if (tx.sender === this.cashlink!.address.toUserFriendlyAddress()) resolve();
+                    // window.setTimeout(reject, 10 * 1000); // 10 seconds timeout
+                });
+            });
+
+            await CashlinkStore.Instance.put(this.cashlink!);
+
+            // Show success screen and redirect to Safe
+            this.statusState = StatusScreen.State.SUCCESS;
+            this.statusTitle = 'Cashlink claimed!';
+
+            window.setTimeout(() => this.redirectToSafe(), StatusScreen.SUCCESS_REDIRECT_DELAY);
+        } catch (err) {
+            // Return to regular screen
+            this.statusState = false;
+        }
+    }
+
+    private selectRecipient() {
+        this.shownAccountSelector = true;
+    }
+
+    private accountSelected(walletId: string, userFriendlyAddress: string) {
+        this.$setActiveAccount({ walletId, userFriendlyAddress });
+        this.shownAccountSelector = false;
+    }
+
+    private goToSignup() {
+        this.callHub('signup');
+    }
+
+    private goToLogin() {
+        this.callHub('login');
+    }
+
+    private goToOnboarding() {
+        this.callHub('onboard');
+    }
+
+    private callHub(method: string) {
+        const request: BasicRequest = {
+            appName: 'Cashlink',
+        };
+
+        const redirectBehavior = new HubApi.RedirectRequestBehavior(undefined, { url: window.location.href });
+        const hubApi = new HubApi(undefined, redirectBehavior);
+
+        switch (method) {
+            case 'signup': hubApi.signup(request); break;
+            case 'login': hubApi.login(request); break;
+            case 'onboard': hubApi.onboard(request); break;
+        }
+    }
+
+    private redirectToSafe() {
+        window.location.href = Config.network === NETWORK_MAIN
+            ? 'https://safe.nimiq.com'
+            : 'https://safe.nimiq-testnet.com';
+    }
+
+    private get isCashlinkStateKnown(): boolean {
+        if (!this.consensus) return false;
```

Comment:
Is that enough of a check here? `this.consensus` might just be `'syncing'` or `'lost'`.

Reply:
`this.consensus` is initialized as `null` and only set to a string when the network client is assigned.
However, I found that I can remove consensus tracking in this component as the Cashlink's internal tracking is sufficient.